src/components/PhoneInput/countriesDropdown/helpers/filterCountries.js

```diff
@@ -0,0 +1,39 @@
+export default function filterCountries(query, countries, country) {
+    if (query === '' || query === '+') {
+        const filteredCountries = countries.filter(value => value.isoCode !== country.isoCode);
+        filteredCountries.unshift(country);
+        return filteredCountries;
+    }
+
+    const queryToLower = query.toLowerCase();
+    let isFilteredCountry = false;
+    const filteredCountries =
+        countries.filter(value => {
+            if (!isNaN(query)) {
+                const index = value.countryCode.toString().indexOf(queryToLower);
+                if (index === 0 || index === 1) {
+                    if (value.isoCode === country.isoCode) {
+                        isFilteredCountry = true;
+                        return false;
+                    }
+                    return true;
+                }
+                return false;
+            }
+            if (
+                value.country.toLowerCase().indexOf(queryToLower) !== -1 ||
+                value.isoCode.indexOf(queryToLower) !== -1
+            ) {
+                if (value.isoCode === country.isoCode) {
+                    isFilteredCountry = true;
+                    return false;
+                }
+                return true;
+            }
+            return false;
+        }) || [];
+    if (isFilteredCountry) {
+        filteredCountries.unshift(country);
+    }
+    return filteredCountries;
```

Comment:
take look here: `src/components/InternalDropdown/helpers/searchFilter.js`
this is a filter function that we use in other projects, maybe we can do similar and simplify this